contrib/swagger.yaml

```diff
@@ -0,0 +1,439 @@
+        - name: snippet
+          in: body
+          schema:
+            $ref: '#/definitions/Snippet'
+      responses:
+        201:
+          schema:
+            $ref: '#/definitions/Snippet'
+          description: Created. A new Snippet has been successfully created.
+        400:
+          schema:
+            $ref: '#/definitions/Error'
+          description: Bad request.
+        401:
+          schema:
+            $ref: '#/definitions/Error'
+          description: |
+            Unauthorized. Returned when a user tries to create a non-anonymous
+            snippet without authorizing first.
+        403:
+          schema:
+            $ref: '#/definitions/Error'
+          description: |
+            Forbidden. Returned when an authorized user tries to create a snippet
+            on behalf of another user.
+
+  /snippets/{id}:
+    get:
+      summary: Retrieve information about the specific code snippet.
+      description: |
+        The Snippet endpoint returns information about the code snippet uniquely
+        identified by the given id value.
+      parameters:
+        - name: id
+          type: integer
+          minimum: 1
+          format: int64
+          required: true
+          in: path
+          description: Unique identifier representing the specific code snippet.
+      responses:
+        200:
+          description: A Snippet object
+          schema:
+            $ref: '#/definitions/Snippet'
+        400:
+          schema:
+            $ref: '#/definitions/Error'
+          description: |
+            Bad request.
+        401:
+          schema:
+            $ref: '#/definitions/Error'
+          description: |
+            Unauthorized. User tries to retrieve a private snippet without
+            authorizing first.
+        403:
+          schema:
+            $ref: '#/definitions/Error'
+          description: |
+            Forbidden. User tries to retrieve a private snippet by another author.
+        404:
+          schema:
+            $ref: '#/definitions/Error'
+          description: Snippet not found.
+    patch:
+      summary: Update an existing code snippet.
+      description: |
+          Update an existing code snippet, possibly creating a new changeset.
+      parameters:
+        - name: id
+          type: integer
+          minimum: 1
+          format: int64
+          required: true
+          in: path
+          description: Unique identifier representing the specific code snippet.
+        - name: snippet
+          in: body
+          schema:
+            $ref: '#/definitions/Snippet'
+          required: true
+          description: |
+              A snippet object, which will be used for update. Can have all or
+              only some of the fields set. Only set fields will be used for
+              update. If new ``content`` is provided, a new changeset of the
+              snippet will be created and set as the current one.
+      responses:
+        201:
```

Comment:
Hm.. `201` means `CREATED`. Do we really want to return that status? What do you think about '200 OK' ?